contrib_bots/lib/john.py

```diff
@@ -0,0 +1,132 @@
+import json
+import os
+import time
+import sys
+
+from random import choice
+
+try:
+    from chatterbot import ChatBot
+    from chatterbot.trainers import ChatterBotCorpusTrainer, ListTrainer
+except ImportError:
+    raise ImportError("""It looks like you are missing chatterbot.
+                      Please: pip install chatterbot""")
+
+CONTRIB_BOTS_DIR = os.path.dirname(os.path.abspath('__filename__'))
```

Comment:
`'__filename__'` is just a string and has no special meaning. You probably want to use the `__file__` variable.